cmd/init.go

```diff
@@ -38,180 +46,279 @@ func getInitCommand() *cobra.Command {
 
 			colonyApi := colony.New(apiURL, apiKey)
 			if err := colonyApi.ValidateAPIKey(ctx); err != nil {
-				return fmt.Errorf("error validating api key: %w", err)
+				return fmt.Errorf("error validating colony api key: %q %w \n visit https://colony.konstruct.io to get a valid api key", apiKey, err)
+			}
+
+			log.Info("colony api key provided is valid")
+
+			dockerCLI, err := docker.New(log)
+			if err != nil {
+				return fmt.Errorf("error creating docker client: %w", err)
+			}
+
+			pwd, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("error getting current working directory: %w", err)
 			}
 
-			log.Info("initializing colony cloud with api key:", apiKey)
+			err = dockerCLI.CreateColonyK3sContainer(ctx, loadBalancerIP, loadBalancerInterface, pwd)
+			if err != nil {
+				return fmt.Errorf("error creating container: %w", err)
+			}
+
+			// TODO hack, the kube api is not always ready need to figure out a better condition
+			time.Sleep(time.Second * 7)
 
-			k8sClient, err := k8s.New(log, "/home/vagrant/.kube/config")
+			k8sClient, err := k8s.New(log, filepath.Join(pwd, constants.KubeconfigHostPath))
 			if err != nil {
 				return fmt.Errorf("error creating Kubernetes client: %w", err)
 			}
 
+			coreDNSDeployment, err := k8sClient.ReturnDeploymentObject(
+				ctx,
+				"kubernetes.io/name",
+				"CoreDNS",
+				"kube-system",
+				50,
+			)
+			if err != nil {
+				return fmt.Errorf("error finding coredns deployment: %w", err)
+			}
+
+			k8sClient.WaitForDeploymentReady(ctx, coreDNSDeployment, 120)
+
+			// Create the secret
+			apiKeySecret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "colony-api",
+					Namespace: "tink-system",
+				},
+				Data: map[string][]byte{
+					"api-key": []byte(apiKey),
+				},
+			}
+
 			// Create a secret in the cluster
-			if err := k8sClient.CreateAPIKeySecret(ctx, apiKey); err != nil {
+			if err := k8sClient.CreateSecret(ctx, apiKeySecret); err != nil {
 				return fmt.Errorf("error creating secret: %w", err)
 			}
 
-			log.Info("Applying tink templates")
-			templates, err := colonyApi.GetSystemTemplates(ctx)
+			k8sconfig, err := ioutil.ReadFile(constants.KubeconfigHostPath)
```

Comment:
```suggestion
			k8sconfig, err := os.ReadFile(constants.KubeconfigHostPath)
```

The `ioutil` package is deprecated.